File: src/main/java/com/bot/commands/voice/ShuffleCommand.java
package com.bot.commands.voice;

import com.bot.commands.VoiceCommand;
import com.bot.voice.VoiceSendHandler;
import com.jagrosh.jdautilities.command.CommandEvent;

public class ShuffleCommand extends VoiceCommand {

    public ShuffleCommand() {
        this.name = "shuffle";
        this.help = "Shuffles the current voice queue";
    }

    @Override
    protected void executeCommand(CommandEvent commandEvent) {
        VoiceSendHandler handler = (VoiceSendHandler) commandEvent.getGuild().getAudioManager().getSendingHandler();
        if (handler == null)
            return;
        try {
            handler.shuffleTracks();
            commandEvent.reactSuccess();
        } catch (Exception e) {
            commandEvent.replyError("Failed to shuffle tracks");
            logger.severe("Failed to shuffle tracks", e);
        }

    }
}

File: src/main/java/com/bot/commands/voice/ShuffleCommand.kt
package com.bot.commands.voice

import com.bot.commands.VoiceCommand
import com.bot.voice.VoiceSendHandler
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class ShuffleCommand : VoiceCommand() {
    init {
        name = "shuffle"
        help = "Shuffles the current voice queue"
    }

    @Trace(operationName = "executeCommand", resourceName = "Shuffle")
    override fun executeCommand(commandEvent: CommandEvent) {
        val voiceConnection = guildVoiceProvider.getGuildVoiceConnection(commandEvent.guild)
        try {
            voiceConnection.shuffleTracks()
            commandEvent.reactSuccess()
        } catch (e: Exception) {
            commandEvent.replyError("Failed to shuffle tracks")
            logger.severe("Failed to shuffle tracks", e)
        }
    }
}